src/fmeval/eval_algorithms/qa_accuracy.py

```diff
@@ -238,145 +286,69 @@ class QAAccuracy(EvalAlgorithmInterface):
     eval_name = EvalAlgorithm.QA_ACCURACY.value
 
     def __init__(self, eval_algorithm_config: QAAccuracyConfig = QAAccuracyConfig()):
-        """Default constructor
+        """QAAccuracy initializer.
 
-        :param eval_algorithm_config: QA Accuracy eval algorithm config.
+        :param eval_algorithm_config: QA Accuracy evaluation algorithm config.
         """
         super().__init__(eval_algorithm_config)
         self._eval_algorithm_config = eval_algorithm_config
+        self.transform = QAAccuracyScores(target_output_delimiter=eval_algorithm_config.target_output_delimiter)
+
+    def evaluate_sample(self, target_output: str, model_output: str) -> List[EvalScore]:
+        """Compute QA accuracy metrics for a single sample.
+
+        :param target_output: The expected/desired model output.
+        :param model_output: The actual model output.
+        :returns: A list of EvalScore objects, one for each of the QA accuracy metrics.
+        """
+        target_output_key = self.transform.target_output_key
+        model_output_key = self.transform.model_output_key
+        sample = {target_output_key: target_output, model_output_key: model_output}
+        pipeline = TransformPipeline([self.transform])
```

Comment:
it feels this is making `evaluate_sample` more complicated, was this concerned raised earlier? I understand we are too late to discuss on this now, but can you share the conclusion or thread if a discussion on this was done earlier?

Reply:
It does make things slightly more complicated, but I think it's best to keep things consistent across all algos, where it's clear that `evaluate_sample` and `evaluate` execute the algo's pipeline. Plus, in terms of raw lines of code, there's hardly any increase.

Reply:
one of the major tenet for `evaluate_sample` has been to keep it simple and readable. I feel we are moving against that by adding transforms in the method. Yeah, let's capture a backlog SIM for it, we can discuss with the team then and take a call. 

I am in favor of not having so many hops for customer in `evaluate_sample` code.